### PR TITLE
[MAINTENANCE] ruff `0.3.7`

### PR DESCRIPTION
**Pull request by Kilo59**
* Update from ruff `0.3.5` -> `0.3.7`
  * Applied auto-fix for new rules
* Add option to use `--unsafe-fixes` with our `invoke lint` task.
* Make `invoke fix` use `--unsafe-fixes` by default.

https://github.com/astral-sh/ruff/releases/tag/v0.3.7
https://github.com/astral-sh/ruff/releases/tag/v0.3.6

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -145,16 +147,22 @@ def lint(
...
-    format(ctx, path=path, sort_=True)
+@invoke.task(help={"path": _PATH_HELP_DESC, "safe-only": "Only apply 'safe' fixes."})
+def fix(ctx: Context, path: str = ".", safe_only: bool = False):
+    """
+    Automatically fix all possible code issues.
+    Applies unsafe fixes by default.
+    """
+    unsafe_fixes = not safe_only
+    lint(ctx, path=path, fmt_=False, fix=True, unsafe_fixes=unsafe_fixes)
+    format(ctx, path=path, check=False, sort_=False)
```
Comment (Kilo59) [inline tasks.py]: I've made `invoke fix` automatically apply "unsafe" fixes.

File: great_expectations/core/expectation_diagnostics/expectation_doctor.py [hunk 2]
```diff
@@ -340,7 +340,7 @@ def _get_coverage_score(
         _total_passed = 0
         _total_failed = 0
         _num_backends = 0
-        _num_engines = sum([x for x in execution_engines.values() if x])
```
Comment (NathanFarmer) [inline great_expectations/core/expectation_diagnostics/expectation_doctor.py:343]: TIL that `expectation_doctor` is a thing 😄

**Review by NathanFarmer — COMMENTED** [review]
Maybe I missed a discussion on this, but why would we want to do "unsafe" fixes?

**Review by NathanFarmer — APPROVED** [review]
Nevermind, found the link

**Comment by Kilo59** [comment]
@NathanFarmer my thinking is that if you are explicitly invoking a `fix` command you are attempting to apply any available fixes.

The `unsafe-fix` though are never applied automatically by pre-commit or ci.

When we first added `invoke fix` there was no notion of safe/unsafe fixes.

File: tasks.py [hunk 3]
```diff
@@ -145,16 +147,23 @@ def lint(
...
 
 
-@invoke.task(help={"path": _PATH_HELP_DESC})
-def fix(ctx: Context, path: str = "."):
-    """Automatically fix all possible code issues."""
-    lint(ctx, path=path, fix=True)
-    format(ctx, path=path, sort_=True)
+@invoke.task(help={"path": _PATH_HELP_DESC, "safe-only": "Only apply 'safe' fixes."})
+def fix(ctx: Context, path: str = ".", safe_only: bool = False):
+    """gikt
```
Comment (Kilo59) [inline tasks.py:159]:
```suggestion
    """
```

**Pull request merged**